pkg/cloudprovider/providers/vsphere/vsphere_metrics.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vsphere
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"time"
+)
+
+const (
+	request_createvolume                      = "CreateVolume"
+	request_createvolume_with_policy          = "CreateVolumeWithPolicy"
+	request_createvolume_with_raw_vsan_policy = "CreateVolumeWithRawVSANPolicy"
+	request_deletevolume                      = "DeleteVolume"
+	request_attachvolume                      = "AttachVolume"
+	request_detachvolume                      = "DetachVolume"
+	request_diskIsAttached                    = "DiskIsAttached"
+	request_disksAreAttached                  = "DisksAreAttached"
+)
+
+var vsphereApiMetric = prometheus.NewHistogramVec(
+	prometheus.HistogramOpts{
+		Name: "vsphere_api_request",
```

Comment:
Usually when you plot these metrics, no matter which name you use - it will become obscured. I know it sucks a bit when you list it here in tabular form. 

But more importantly - the actual meaning of these metrics are different from similar metrics from other cloudproviders.